src/backend/core/tests/test_api_documents_search.py

```diff
@@ -13,12 +13,26 @@
 from rest_framework.test import APIClient
 
 from core import enums, factories
+from core.services.opensearch import check_hybrid_search_enabled, opensearch_client
 
-from .utils import build_authorization_bearer, prepare_index, setup_oicd_resource_server
+from .mock import albert_embedding_response
+from .utils import (
+    build_authorization_bearer,
+    bulk_create_documents,
+    prepare_index,
+    setup_oicd_resource_server,
+)
 
 pytestmark = pytest.mark.django_db
 
 
+@pytest.fixture(autouse=True)
+def clear_caches():
+    """Clear cached functions before each test to avoid side effects"""
+    opensearch_client.cache_clear()
+    check_hybrid_search_enabled.cache_clear()
```

Comment:
I think this might generate a side effect, because:

- cache is cleared
- setting is updated in test
- new value is computed and put in cache
- test ends
- setting is rolled back to previous value
- 💥 cache is not cleared again

Several solutions:

- Put this fixture in a conftest at the root, so every tests use it: side effect, the value is "computed every time" but it not a problem
- Improve the fixture to store the original value, clean the cache, run the test, put the original value back, clean the cache again (or simply: clean the cache, run the test, clean the cache, this should also work)